sw/host/opentitanlib/src/transport/hyperdebug/mod.rs

```diff
@@ -153,6 +158,54 @@ impl Hyperdebug {
                             }
                         }
                     }
+                    "I2C" => {
+                        // We found the I2C forwarding USB interface (this one interface allows
+                        // multiplexing physical I2C ports.)
+                        let mut in_endpoint: Option<u8> = None;
+                        let mut out_endpoint: Option<u8> = None;
+                        for endpoint_desc in interface_desc.endpoint_descriptors() {
+                            if endpoint_desc.transfer_type() != rusb::TransferType::Bulk {
+                                continue;
+                            }
+                            match endpoint_desc.direction() {
+                                rusb::Direction::In => {
+                                    if let Some(_) = in_endpoint.replace(endpoint_desc.address()) {
+                                        return Err(Error::CommunicationError(
```

Comment:
Should this be an `ensure!(foo.is_none(), ...`?

Reply:
Done, here and in the existing SPI section.